examples/bri-3/src/bri/workgroup/workgroups/capabilities/archiveWorkgroup/archiveWorkgroupCommand.handler.ts

```diff
@@ -0,0 +1,25 @@
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { WorkgroupAgent } from '../../agents/workgroups.agent';
+import { WorkgroupStorageAgent } from '../../agents/workgroupStorage.agent';
+import { ArchiveWorkgroupCommand } from './archiveWorkgroup.command';
+
+@CommandHandler(ArchiveWorkgroupCommand)
+export class ArchiveWorkgroupCommandHandler
+  implements ICommandHandler<ArchiveWorkgroupCommand>
+{
+  constructor(
+    private workgroupAgent: WorkgroupAgent,
+    private workgroupStorageAgent: WorkgroupStorageAgent,
+  ) {}
+
+  async execute(command: ArchiveWorkgroupCommand) {
+    const workgroupToArchive =
+      await this.workgroupAgent.fetchUpdateCandidateAndThrowIfUpdateValidationFails(
```

Comment:
Is this validation ok for this use case? We want to majke sure it exists and that the status is ACTIVE. Maybe a separate method is better.